src/kixi/datastore/metadatastore/command_handler.clj

```diff
@@ -180,18 +212,37 @@
                          ::cs/file-metadata-update-type
                          ::cs/file-metadata-created}))))
 
-(defn create-sharing-change-handler
+(defn create-sharing-change-handler-old
   [metadatastore]
   (let [authorised (partial ms/authorised metadatastore ::ms/meta-update)]
     (fn [{:keys [::kc/payload] :as cmd}]
       (cond
-        (not (spec/valid? ::sharing-change-cmd-payload payload)) (sharing-change-invalid cmd)
-        (not (authorised (::ms/id payload) (get-user-groups cmd))) (sharing-change-unauthorised cmd)
+        (not (spec/valid? ::ms/sharing-change-payload payload)) (sharing-change-invalid-old cmd)
+        (not (authorised (::ms/id payload) (get-user-groups cmd))) (sharing-change-unauthorised-old cmd)
         :default (updated (merge {::cs/file-metadata-update-type
                                   ::cs/file-metadata-sharing-updated
                                   :kixi/user (::kc/user cmd)}
                                  payload))))))
 
+(defn get-spec-keys
```

Comment:
This feels like it trying too hard... The keys aren't likely to change, could just select-keys